test/unit/mdc-textfield/mdc-textfield.test.js

```diff
@@ -48,13 +49,54 @@ test('#constructor assigns helptextElement to the id specified in the input aria
   document.body.removeChild(helptext);
 });
 
+class FakeRipple {
+  constructor(root) {
+    this.root = root;
+    this.layout = td.func('.layout');
+    this.destroy = td.func('.destroy');
+  }
+}
+
+test('#constructor when given a `mdc-textfield--box` element instantiates a ripple on the root element', () => {
+  const root = getFixture();
+  root.classList.add(cssClasses.BOX);
+  const component = new MDCTextfield(root, undefined, (el) => new FakeRipple(el));
+  assert.equal(component.ripple.root, root);
+});
+
+test('#constructor sets the ripple property to `null` when given a non `mdc-textfield--box` element', () => {
+  const component = new MDCTextfield(getFixture());
+  assert.isNull(component.ripple);
+});
+
+test('#constructor initializes a default ripple when no ripple factory given when given a ' +
```

Comment:
How about "constructor initializes a default ripple when ripple factory is not given in `mdc-textfield--box` element?"